addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -586,21 +589,18 @@ public synchronized void showNotification(String message) {
 
     public boolean checkConnection() {
         if (!socket.isConnected()) {
-            logger.debug("checkConnection: try to connect to kodi {}", wsUri.toString());
+            logger.debug("checkConnection: try to connect to Kodi {}", wsUri);
             try {
                 socket.open();
                 return socket.isConnected();
-            } catch (Throwable t) {
-                logger.error("exception during connect to {}", wsUri.toString(), t);
-                try {
-                    socket.close();
-                } catch (Exception e) {
-                }
+            } catch (Exception e) {
+                logger.error("exception during connect to {}", wsUri, e);
+                socket.close();
```

Comment:
Can/will closing a not open socket give any errors?

Reply:
No, it won't. All errors will be handled inside the `socket.close()` method.

Reply:
👍